docs/design/adrs/2024-04-23-implement-prompflow-orchestration.md

```diff
@@ -0,0 +1,42 @@
+# Implement a PromptFlow Orchestration as an alternative for the chat backend
+
+* **Status:** proposed
+* **Proposer:** @superhindupur
+* **Date:** 2024-04-23
+* **Technical Story:** [Include a prompt flow integration](https://github.com/Azure-Samples/chat-with-your-data-solution-accelerator/issues/406)
+
+## Context and Problem Statement
+
+The chat functionality in this repo has an orchestraction mechanism for fetching the answer based on the user's question - the question is sent to the LLM, then based on whether it needs to be searched, it goes to the search service before the answer is finally reprocessed and sent back to the user on the chat UI.
+
+This orchestration between the chat backend, LLM and search service is done inside code and currently has two options for orchestration management - openai functions and langchain (controlled by the deployment bicep param `orchestrationStrategy`). In this ADR, we explore whether it is worth adding a third mechanism to orchestrate the chat backend functionality with [Azure Machine Learning PromptFlow](https://learn.microsoft.com/en-us/azure/machine-learning/prompt-flow/overview-what-is-prompt-flow?view=azureml-api-2).
+
+## Decision Drivers
+
+* Being able to demonstrate various orchestration options to customers
+* Limiting complexity in the repo
+* Ease of maintenance of the PromptFlow configuration
+
+## Considered Options
+
+* Adding the PromptFlow orchestration
+* Not adding the PromptFlow orchestration
+
+## Decision Outcome
+In the absence of concrete business / customer requirements, the outcome is to NOT add a PromptFlow orchestrator alternative for the chat backend.
+
+
+## Pros and Cons of the Options
+
+### Adding the PromptFlow orchestration
+* Good, because customers wanting to deploy the chat backend as a PromptFlow endpoint will be able to do so.
+* Good, because customers will have a low-code mechanism to experiment with the orchestration of the chat backend.
+* Bad, because building a PromptFlow orchestration that mimics the current chat backend requires significant effort - and it is unclear how much customer demand we have for it.
```

Comment:
I might be being optimistic, but I don't think it will be significant effort. Is it work breaking it down and doing rough estimates?

Reply:
Fair enough, the "significant effort" is subjective. My suggestion is - let's hold off on this until we're able to validate customer requirements with Randy tomorrow and maybe we will have a strong argument in favor of adding promptflow.